hks-epod/edx-platform
# Backport section navigation improvements to Dogwood

Backports the section navigation improvements merged to master for [PR 11705](https://github.com/edx/edx-platform/pull/11705), including subsequent bug fixes.

**JIRA tickets**: [OC-1574](https://tasks.opencraft.com/browse/OC-1574)

**Screenshots**:

Next button is enabled, even though we've navigated to the end of a section:
![prev_and_next](https://cloud.githubusercontent.com/assets/7556571/14911387/3216a34e-0e34-11e6-9636-b8a6bd909e65.png)

**Sandbox URL**: TBD - @bradenmacdonald should I build a sandbox for this?

**Testing instructions**:
1. Set up a devstack with OPENEDX_RELEASE="named-release/dogwood".
2. Checkout this repo/branch
3. Run LMS: `paver devstack lms`
4. Verify that you can use the courseware prev/next buttons to skip across sections.
5. Run tests: `paver test` 

**Author notes and concerns**:
1. I've tested that the section prev/next feature works in the LMS, but some (unrelated?) automated tests are still failing, probably because my cherry-picked commits aren't complete.  I will address those in the next couple of days.
2. I encountered [this issue with apt sources](https://www.reddit.com/r/chrome/comments/48oje6/linux_how_to_fix_failed_to_fetch/) when setting up a fresh devstack, which affected ansible's ability to install system packages.  The error can be seen by running `apt-get update` on your vagrant instance:
   
   ```
   Failed to fetch http://dl.google.com/linux/chrome/deb/dists/stable/Release
   Unable to find expected entry 'main/binary-i386/Packages' in Release file
   (Wrong sources.list entry or malformed file)
   ```
   
   Had to do both these steps to address it:
   
   ```
   sudo vim /etc/apt/sources.list.d/google-chrome.list  # correct line to include [arch=amd64]:
   # deb [arch=amd64] http://dl.google.com/linux/chrome/deb/ stable main
   sudo chattr +i /etc/apt/sources.list.d/google-chrome.list  # to prevent it from being overwritten
   ```

**Reviewers**
- [ ] other reviewers TBD
